### PR TITLE
Update Namespace's metadata as chart CR

### PR DESCRIPTION
**Pull request by tomahawk28**
Toward https://github.com/giantswarm/giantswarm/issues/16209

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by tomahawk28** [comment]
Tested in `ginger`
```
kubectl -n giantswarm charts app-operator-unique
...
  namespaceConfig:
    annotations:
      testa: testb
    labels:
      t: tt
```


```
apiVersion: v1
kind: Namespace
metadata:
  annotations:
    testa: testb
  creationTimestamp: "2020-01-15T14:08:24Z"
  labels:
    giantswarm.io/managed-by: chart-operator
    name: giantswarm
    t: tt
  name: giantswarm
  resourceVersion: "216241086"
  selfLink: /api/v1/namespaces/giantswarm
  uid: 0cbded45-3ad0-44e6-951a-2ddad5d11eaa
spec:
  finalizers:
  - kubernetes
status:
  phase: Active
```

File: service/controller/chart/resource/namespace/create.go [hunk 1]
```diff
@@ -58,3 +70,50 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
...
+	}
+
+	for k, v := range key.NamespaceAnnotations(cr) {
+		if namespace.GetAnnotations()[k] != v {
+			namespace.GetAnnotations()[k] = v
+			updated = false
+		}
+	}
+
+	if updated {
```
Comment (tomahawk28) [inline service/controller/chart/resource/namespace/create.go:104]: Only if the operator saw there was an update on a map object, we update a namespace object.

File: service/controller/chart/resource/namespace/create.go [hunk 2]
```diff
@@ -58,3 +70,50 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
...
+func (r *Resource) ensureNamespaceUpdated(ctx context.Context, cr v1alpha1.Chart) error {
+	namespace, err := r.k8sClient.CoreV1().Namespaces().Get(ctx, key.Namespace(cr), metav1.GetOptions{})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	updated := true
+
+	if namespace.GetLabels() == nil {
+		namespace.Labels = map[string]string{}
```
Comment (tomahawk28) [inline service/controller/chart/resource/namespace/create.go:83]: To avoid assigning into nil, I'm giving initialized map here.

**Review by rossf7 — APPROVED** [review]
LGTM

**Pull request merged**